### PR TITLE
Remove the warning about `rubocop-performance` from non-related output formats

### PR DESCRIPTION
**Pull request by AlexWayfer**
Fix https://github.com/AtomLinter/linter-rubocop/issues/309 after #6845

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [ ] Added tests.
* [x] Added an entry to the [Changelog](https://github.com/rubocop-hq/rubocop/blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Run `bundle exec rake default`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: lib/rubocop/formatter/simple_text_formatter.rb [hunk 1]
```diff
@@ -23,6 +23,18 @@ class SimpleTextFormatter < BaseFormatter
       def started(_target_files)
         @total_offense_count = 0
         @total_correction_count = 0
+
+        return unless defined?(Bundler)
+
+        gemfile_lock = Bundler.read_file(Bundler.default_lockfile)
+        parser = Bundler::LockfileParser.new(gemfile_lock)
+
+        unless parser.dependencies['rubocop-performance']
```
Comment (vzamanillo) [inline lib/rubocop/formatter/simple_text_formatter.rb]: To fix `Style/GuardClause` on `internal_investigation` tests

```suggestion
      return unless parser.dependencies['rubocop-performance']

      warn Rainbow(<<-MESSAGE.strip_indent).yellow.to_s
        [Warn] Performance Cops will be removed from RuboCop 0.68. Use rubocop-performance gem instead.
               https://github.com/rubocop-hq/rubocop/tree/master/manual/migrate_performance_cops.md
      MESSAGE
```

Comment (AlexWayfer) [reply lib/rubocop/formatter/simple_text_formatter.rb]: Thank you! My bad. Done.

**Comment by bbatsov** [comment]
I don't quite get the fix. Why move this to one particular output format? That'd be almost the same as removing it completely.

**Comment by AlexWayfer** [comment]
> I don't quite get the fix.

Did you read the commit message (PR description)? There is the issue with tools which rely on RuboCop JSON (or another standard) output. The current output in JSON format (`--format=json`) is invalid JSON. I think, the same for HTML or other (quite?) formats. Is this not enough? Have tools to bypass any non-formatted text somehow?

---

New conflicts resolved.

**Comment by bbatsov** [comment]
Why are those tools reading something from stderr? That seems pretty weird.

**Comment by AlexWayfer** [comment]
> Why are those tools reading something from stderr? That seems pretty weird.

Let me try to discuss this in the tool's issue. Thank you for advice.

**Comment by AlexWayfer** [comment]
I think, we can solve it in tools. Thank you, sorry to bother you.

**Pull request closed without merging**